checkprovider

```diff
@@ -191,6 +205,13 @@ for ( my $test = 0 ; !$period || $test < $total ; $total && $test++ ) {
     # for child processes
     if ($pid) {
 
+        # advance the term list position for the next child
+        if ( ($position + 1) > $#terms ) {
```

Comment:
Ya, it was a bit tricky to find the right place for this since I didn't write this part. This only increments the position in the parent process. If it is done in the child then the position value won't persist. It also increments for each child created.